File: lib/ce2/models/subtrack.rb
class Subtrack < Tuple
  attribute :answer_mode_timeout, :integer
  attribute :review_mode_timeout, :integer
  attribute :published_at, :datetime
  attribute :deleted_at, :datetime
  attribute :track_id, :string

  # belongs_to :track
  relates_to_one :track do
    Track.where(Track.id.eq(track_id))
  end

  # has_many :question_sets
  relates_to_many :question_sets do
    QuestionSet.where(QuestionSet.subtrack_id.eq(id))
  end
end

File: javascript/lib/models/subtrack.js
June.define_set("Subtrack", function(c) { with(c) {
  attributes({
    id: "string",
    name: "string",
    answer_mode_timeout: "integer",
    review_mode_timeout: "integer",
    published_at: "datetime",
    deleted_at: "datetime",
    track_id: "string"
  });

  belongs_to("track");
  has_many("question_sets")
}});